data.go

```diff
@@ -6,17 +6,28 @@ import (
 
 // SMTPError specifies the error code and message that needs to be returned to the client
 type SMTPError struct {
-	Code    int
-	Message string
+	Code         int
+	EnhancedCode [3]int
+	Message      string
 }
 
+// NoEnhancedCode is used to indicate that enhanced error code should not be
+// included in response.
+var NoEnhancedCode = [3]int{-1, -1, -1}
+
+// EnhancedCodeNotSet is a nil value of EnhancedCode field in SMTPError, used
+// to indicate that backend failed to provide enhanced status code. X.0.0 will
+// be used (X is derived from error code).
+var EnhancedCodeNotSet = [3]int{0, 0, 0}
```

Comment:
Should this be the default for `SMTPError`? Or should `NoEnhancedCode` be the default?

Reply:
`NoEnhancedCode` should be only used in special cases permitted by protocol (such as EHLO response). It is exported for use by extensions. 

On the other hand, RFC 2034 requires enhanced code to be included in all 2xx, 4xx and 5xx responses. I don't want to break clients expectations here because of incomplete backend implementation. Hence EnhancedCodeNotSet is used by default, kinda similar to fallback updates implementation in go-imap if backend doesn't implement the Updater interface.

Reply:
Makes sense